hive_metastore_client/hive_metastore_client.py

```diff
@@ -131,7 +131,11 @@ def add_partitions_to_table(
             table_partition_keys=table.partitionKeys,
         )
 
-        self.add_partitions(partition_list_with_correct_location)
+        try:
+            self.add_partitions(partition_list_with_correct_location)
```

Comment:
what if we checked if the partitions_list is `not empty` or `None`?

Reply:
We can do it, but I have a concern: in this case, we'd return False and the user wouldn't know the reason the method returned false unless we add a message in the method's return (since we don't log inside the lib).
The other option (that is the current behavior) is to let the user validate the arguments she/he is sending to this method. And if she/he sends an invalid value an exception will be naturally thrown by this client.
What do you think?

Reply:
yeah, I understand =]
I also think returning False would get confusing.

my point is more related to the function knowing how to handle parameters and to throw specific exception in the case the caller passes a wrong/invalid value.
because leaving this responsibility to the caller would make us replicate the same logic for multiple callers, and if keep it inside the function that's gonna do something with it, it would be centralized.

and, in this case, I believe a `raise ValueError` in the beginning would do the trick (instead of returning False)

makes sense? what do you think?

Reply:
I think it's a good idea to raise an exception if the user sends an invalid/empty value here. :+1:

Reply:
Done: https://github.com/quintoandar/hive-metastore-client/pull/37/commits/a2ba8c8cc09bd80a71b1066b31953b938d442b10